lib/instrumentation/transaction.js

```diff
@@ -218,3 +223,76 @@ Transaction.prototype.end = function (result, endTime) {
   this._agent._instrumentation.addEndedTransaction(this)
   this._agent.logger.debug('ended transaction %o', { trans: this.id, parent: this.parentId, trace: this.traceId, type: this.type, result: this.result, name: this.name })
 }
+
+Transaction.prototype._captureBreakdown = function (span) {
+  const agent = this._agent
```

Comment:
Not sure how worthwhile it is, but you could make a minor optimisation here by returning if `this.ended`, moving `this.ended = true` directly after the call to `this._captureBreakdown(this)` in `Transaction.end`. That way spans that end after the transaction won't accumulate anything in the transaction.

Reply:
Good call, I'll make a change for that later. Just on my way to give a talk now.